smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    /**
+     * Generates zero or more @see ValidationEvents and returns them in a collection.
+     *
+     * @param occurrence text occurrence found in the body of the model
+     */
+    private Collection<ValidationEvent> getValidationEvents(TextInstance instance) {
+        final Collection<ValidationEvent> events = new ArrayList<>();
+        for (Map.Entry<String, List<String>> termEntry : termsMap.entrySet()) {
+            final String termLower = termEntry.getKey().toLowerCase();
+            final int startIndex = instance.getText().toLowerCase().indexOf(termLower);
+            if (startIndex != -1) {
+                final String matchedText = instance.getText().substring(startIndex, startIndex + termLower.length());
+                switch (instance.getLocationType()) {
+                    case NAMESPACE:
+                        //Cannot use any warning() overloads because there is no shape associated with the event.
+                        events.add(ValidationEvent.builder()
+                                .sourceLocation(SourceLocation.none())
+                                .id(this.getClass().getSimpleName().replaceFirst("Validator$", ""))
+                                .severity(Severity.WARNING)
+                                .message(formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance))
+                                .build());
+                        break;
+                    case APPLIED_TRAIT:
+                        events.add(warning(instance.getShape(),
+                                instance.getTrait().getSourceLocation(),
+                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
+                        break;
+                    case SHAPE:
+                    default:
+                        events.add(warning(instance.getShape(),
+                                instance.getShape().getSourceLocation(),
+                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
+                }
+            }
+        }
+        return events;
+    }
+
+    private static String formatNonInclusiveTermsValidationMessage(
+            Map.Entry<String, List<String>> termEntry,
+            String matchedText,
+            TextInstance instance) {
+        final List<String> caseCorrectedEntryValue = termEntry.getValue().stream()
+            .map(replacement -> Character.isUpperCase(matchedText.charAt(0))
+                  ? StringUtils.capitalize(replacement)
+                  : StringUtils.uncapitalize(replacement))
+            .collect(Collectors.toList());
+        String replacementAddendum = !termEntry.getValue().isEmpty()
+                ? String.format(" Consider using one of the following terms instead: %s",
+                    ValidationUtils.tickedList(caseCorrectedEntryValue))
+                : "";
+        switch (instance.getLocationType()) {
```

Comment:
I see what you're saying. This whole thing breaks down to:

```
   builder.append("{textLocationSymbol}");
   builder.append("{locationType} {TypeDependentMessage}");
   builder.append("contains/uses a non-inclusive term {nonInclusiveTerm}.");
   builder.append("{replacementAddendum}");
```

It would create a top level section of a code that reads cleaner, but require considerably more code as I would need 3 switches to evaluate those components versus one which calls format() differently. I wouldn't want to put message formatting logic in TextInstance.TextLocationType methods to avoid the switch-case statements because I consider TextInstance.TextLocationType agnostic to this particular validator's event message formatting.